### PR TITLE
feat(site): Contact form (accessibility)

### LINKED ISSUE #520: feat: Contact form
We need a contact form for answering questions. Requirements:

 

- [x] A page should live at `/contact` that pulls in content from Contentful snippets. I created a stunt copy one with the slug `contact-page`. You can see examples of using a snippet in `/src/pages/data/index.js` - there's a GraphQL query to grab a snippet from Contentful (query name `dataPreamble`), then we put it in the body of the page.
- [x]  Turn the arrow "Want to get involved?" sections of the homepage into reusable components
- [x]  Add one to the bottom of the page that links to `/contact/accessibility`, the text can be 
- [x]  "Report a problem with accessibility on the website."
- [x] Create a page a `/contact/accessibility`
- [x] Add a form with two fields, "What is your email address?" and "What accessibility problems are you experiencing?"
- [x] Make sure the `<form>` tag has the attribute `data-netlify="true"`
- [x] Implement [spam prevention from Netlify](https://docs.netlify.com/forms/spam-filters/#honeypot-field) on the form.

Once this is set up, we can use the Form in a deploy preview and test it in Slack.

### PR DESCRIPTION
**Pull request by friej715**
Fixes #520 

First pass at contact form! (Also turns Get Involved bullet list into a component.)

I thought it might be handy to make the contact form a component, since we have other contact reasons we want to use it for--but it does assume that any form is just an email address and a text area.

Also holler if there are too many hard numbers in the scss--tried to leverage our own variables (breakpoints, colors, etc.) where possible.

### CODE REVIEW

File: src/pages/contact/accessibility.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import Layout from '../../components/layout'
+import ContactForm from '../../components/common/contact-form'
+
+const ContactAccessibilityPage = () => (
+  <Layout title="Contact us &mdash; Accessibility">
+    <div>
+      Please fill out the form below to contact us about accessibility issues.
+      All fields are required.
+    </div>
+    <ContactForm messageCopy="What accessibility problems are you experiencing?" />
```
Comment (kevee) [inline src/pages/contact/accessibility.js:11]: Let's add a `name="accessibility"` prop here

File: src/components/common/contact-form.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import ContactFormStyles from './contact-form.module.scss'
+
+export default ({ messageCopy }) => {
```
Comment (kevee) [inline src/components/common/contact-form.js:4]: Let's accept a `name` prop

File: src/components/common/contact-form.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  return (
+    <form
+      className={ContactFormStyles.contactForm}
+      name="contact"
```
Comment (kevee) [inline src/components/common/contact-form.js:8]: Change the name attribute to use the `name` prop

File: src/components/common/contact-form.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      method="POST"
+      data-netlify-recaptcha="true"
+      data-netlify="true"
+    >
+      <div>
```
Comment (kevee) [inline src/components/common/contact-form.js:13]: Netlify also requires a hidden input element with Gatsby, which I just learned and isn't documented anywhere. Let's add:

```
<input type="hidden" name="form-name" value={name} />`

**Comment by friej715** [comment]
Copy that, @kevee! I think this last commit should be good to go.

**Comment by kevee** [comment]
Thanks! I'll check it out tomorrow morning.

**Comment by kevee** [comment]
The form isn't submitting, I'm testing in Netlify in a different site and will get back to you.

**Comment by kevee** [comment]
I made a change to using a honeypot instead of recaptcha. The form submissions worked!

More things that need to be made before we can merge:

- [ ] Create a simple "Success" page, and change form actions to go there. It can live at `/contact/success`:

> **Thank you for contacting us!**
> We will try our best to get in touch shortly.

- [ ] Change the link for contacting us in the accessibility page to the accessibility contact form
- [ ] Instead of having to make multiple "contact" pages, lets automate it and put an array of forms into our config file:
1. Create a "contact form" array in `gatsby-config.js`:
```javascript
module.exports = {
  siteMetadata: {
    
    ...stuff...
    contactForms: [
      {
        name: 'accessibility',
        title: 'Accessibility',
        preamble: 'Please fill out the form below to contact us about accessibility issues. All fields are required.',
        message: 'What accessibility problems are you experiencing?'
      }
    ],
]
```
2. Add a query to the `gatsby-node.js` file to query that list of forms and build them using templates, just like we do with the state pages, etc.

File: src/components/common/contact-form.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      netlify-honeypot="covid-bot-field"
+      data-netlify="true"
+    >
+      <div>
+        <input type="hidden" name="form-name" value={name} />
+        <label htmlFor="name">
+          <p className={ContactFormStyles.inputHeader}>
+            What is your email address?
+          </p>
+          <input type="email" id="name" name="name" required />
```
Comment (schwartzadev) [inline src/components/common/contact-form.js:20]: Is there a way we could make the `id` / `name` attrs be equal to `email` or similar? `name` seems like a confusing label for an email field.

File: src/components/common/contact-form.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      name={name}
+      method="POST"
+      action="/contact/success"
+      netlify-honeypot="covid-bot-field"
+      data-netlify="true"
+    >
+      <div>
+        <input type="hidden" name="form-name" value={name} />
+        <label htmlFor="name">
+          <p className={ContactFormStyles.inputHeader}>
```
Comment (schwartzadev) [inline src/components/common/contact-form.js:17]: Can we use `label` here and associate the labels with the inputs with `for`?

File: src/components/common/contact-form.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            What is your email address?
+          </p>
+          <input type="email" id="name" name="name" required />
+        </label>
+      </div>
+      <div>
+        <label htmlFor="message">
+          <p className={ContactFormStyles.inputHeader}>{messageCopy}</p>
```
Comment (schwartzadev) [inline src/components/common/contact-form.js:25]: See above about `label`, `for`

File: src/components/layout/footer.js [hunk 8]
```diff
@@ -11,13 +11,16 @@ const Footer = () => (
         <img src={logo} alt="The COVID Tracking Project" />
         <ul>
           <li>
-            <a href="https://github.com/COVID19Tracking">GitHub</a>
+            <a href="/contact">Contact us</a>
```
Comment (schwartzadev) [inline src/components/layout/footer.js:14]: I _think_ we would want this to be `Contact Us` (capital `U`), to align with the caps in the nav/footer

File: src/pages/contact.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import { Link, graphql } from 'gatsby'
+import Layout from '../components/layout'
+import GetInvolved from '../components/common/get-involved'
+
+const ContactPage = ({ data }) => (
+  <Layout title="Contact us" narrow textHeavy>
```
Comment (schwartzadev) [inline src/pages/contact.js:7]: See caps comment above

File: src/components/common/contact-form.module.scss [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    }
+  }
+
+  .input-header {
+    margin-bottom: 5px;
+    font-weight: 600;
+    font-size: 24px;
+  }
+
+  button {
```
Comment (schwartzadev) [inline src/components/common/contact-form.module.scss:37]: Currently, there's a little border for me (FF on Windows):

image

I would go with:
- `border: 1.5px solid $color-blueberry-400;` (or similar) or
- `border: none;`

File: src/components/common/contact-form.module.scss [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    width: 210px;
+    height: 58px;
+    background: $color-blueberry-200;
+    border-radius: 4px;
+    line-height: 33px;
+    color: $color-infobox-sync-dark;
```
Comment (schwartzadev) [inline src/components/common/contact-form.module.scss:43]: I would opt for one of the blueberry/plum colors here. (The infobox colors are not _really_ part of the palette -- that's my bad for causing that confusion.) You might be able to use the existing `Button` component with the form, too.

File: src/pages/contact/success.js [hunk 12]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+import Layout from '../../components/layout'
+
+export default () => (
+  <Layout title="Thank you" narrow textHeavy>
+    <h2>Thank you for contacting us!</h2>
+    <p>We will try our best to get in touch shortly.</p>
```
Comment (schwartzadev) [inline src/pages/contact/success.js:7]: Our accessibility statement provides an explicit timeline: `We try to respond to feedback within 5 business days, and to propose a solution within 10 business days.` I think it makes sense to put that here for the accessibility form.

File: src/components/common/contact-form.js [hunk 13]
```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import ContactFormStyles from './contact-form.module.scss'
+
+export default ({ name, messageCopy }) => {
```
Comment (schwartzadev) [inline src/components/common/contact-form.js:4]: I _love_ the reusability here!

File: src/pages/index.js [hunk 14]
```diff
@@ -39,22 +40,20 @@ export default ({ data }) => (
...
-              </p>
-            </div>
-            <div className={homepageStyles.getInvolved}>
-              <div className={homepageStyles.getInvolvedIcon}>→</div>
-              <p>
-                Want to get involved?{' '}
-                <Link to="/help">Help us get better data</Link>.
-              </p>
-            </div>
+            <GetInvolved
```
Comment (schwartzadev) [inline src/pages/index.js:43]: Great call to make this into a component :raised_hands:

File: src/components/common/contact-form.js (hunk 6 above)
Comment (friej715) [reply src/components/common/contact-form.js:17]: Do you mean something like:

```
        <label htmlFor="email" className={ContactFormStyles.inputHeader}>
          What is your email address?
          <input type="email" id="email" name="email" required />
        </label>
```

File: src/components/common/contact-form.module.scss (hunk 11 above)
Comment (friej715) [reply src/components/common/contact-form.module.scss:43]: Ooh, reusing that component is a good idea.

It looks like ESLint doesn't want React button types to be dynamically defined, so unfortunately I don't believe I can just go `type={buttonType}`. Does the below seem reasonable? (Should it include the `reset` case too...?) Trying not break anything that already uses Button.

```
const Button = ({ big, children, type = 'button' }) => {
  if (type === 'submit') {
    return (
      <button className={`button${big ? ' button-big' : ''}`} type="submit">
        {children}
      </button>
    )
  }
  return (
    <button className={`button${big ? ' button-big' : ''}`} type="button">
      {children}
    </button>
  )
}```

File: src/pages/contact/success.js (hunk 12 above)
Comment (friej715) [reply src/pages/contact/success.js:7]: Makes sense. Do we think that copy will change across form types? (Accessibility, volunteering, general feedback, etc. etc.) Right now it's one success page, so if the copy changes from form to form, we'll have to pass the form type in, right?

File: src/components/layout/footer.js (hunk 8 above)
Comment (kevee) [reply src/components/layout/footer.js:14]: I say just "Contact"

File: src/components/common/contact-form.module.scss (hunk 11 above)
Comment (kevee) [reply src/components/common/contact-form.module.scss:43]: Updating the component is fine with me.

File: src/pages/contact/success.js (hunk 12 above)
Comment (kevee) [reply src/pages/contact/success.js:7]: Let's not get into a dynamic success page, this page should be used by all the forms. How about we add that timeline to the preamble of `/src/pages/contact/accessibility.js`

File: src/components/common/contact-form.js (hunk 6 above)
Comment (schwartzadev) [reply src/components/common/contact-form.js:17]: Yes! Perfect.

Relatedly, I'm not sure if we need the `id` attrs...is that for Netlify?

Comment (friej715) [reply src/components/common/contact-form.js:17]: Nope, good catch! The only thing we need for Netlify here is that hidden input, I think.

**Comment by friej715** [comment]
One sec and I'll fix the conflict.

**Comment by friej715** [comment]
I thiiiiiiink this addresses everything. Working on a few things at once so pardon any distraction-mistakes 😅

File: src/components/layout/footer.js [hunk 15]
```diff
@@ -13,13 +13,16 @@ const Footer = () => (
         </Link>
         <ul>
           <li>
-            <a href="https://github.com/COVID19Tracking">GitHub</a>
+            <a href="/contact">Contact</a>
```
Comment (kevee) [inline src/components/layout/footer.js:16]: Let's use a `<Link` component instead of an a

**Pull request merged**